src/module/system/text-editor.ts

```diff
@@ -347,6 +350,133 @@ class TextEditorPF2e extends TextEditor {
+            if (glyph) {
+                element.dataset["pf2Glyph"] = glyph;
+            }
+
+            // options
+            const params = options.split(/\s+/).reduce(
+                (result, option) => {
+                    const [key, value] = option.split("=").map((s) => s.trim());
+                    result[key] = value;
+                    return result;
+                },
+                {} as Record<string, string>,
+            );
+
+            // attributes
+            const aliases: Record<string, string> = {
+                ["difficulty-class"]: "dc",
+                stat: "skill",
+                statistic: "skill",
+            };
+            for (const [alias, value] of Object.entries(params)) {
+                const key = aliases[alias] ?? alias;
+                const attribute = sluggify(`pf2-${key}`, { camel: "dromedary" });
+                element.dataset[attribute] = value;
+            }
+
+            // variant
+            const variant = action.variants.get(params.variant?.trim() ?? "");
+
+            if (label?.trim()) {
+                // label
+                element.innerText = label.trim();
+            } else {
+                // name
+                const text = document.createElement("span");
+                text.innerText = variant?.name
+                    ? `${game.i18n.localize(action.name)} - ${game.i18n.localize(variant.name)}`
+                    : game.i18n.localize(action.name);
+                element.appendChild(text);
+
+                // difficulty class
+                const dc = params["difficulty-class"] || params["dc"];
+                const visibility = (params["show-dc"] || "all").trim().toLowerCase();
+                const showDC =
+                    (visibility === "all" && game.pf2e.settings.metagame.dcs) ||
+                    (["all", "gm"].includes(visibility) && game.user.isGM);
+
+                // statistic
+                const statistic = (
+                    params["statistic"] ||
+                    params["stat"] ||
+                    params["skill"] ||
+                    (variant && "statistic" in variant && typeof variant.statistic === "string"
+                        ? variant.statistic
+                        : "") ||
+                    ("statistic" in action && typeof action.statistic === "string" ? action.statistic : "")
+                )?.trim();
+
+                if ((dc && showDC) || statistic) {
+                    const STATISTIC_LABELS: Record<string, string> = {
+                        perception: "PF2E.PerceptionLabel",
+                        ...CONFIG.PF2E.saves,
+                        ...CONFIG.PF2E.skillList,
+                        unarmed: "PF2E.TraitUnarmed",
+                    };
+                    element.appendChild(document.createTextNode(" "));
+
+                    const details = document.createElement("span");
+                    if (dc && showDC && Number.isNumeric(dc)) {
```

Comment:
We should have our own helper for this: mind making something like `isIntegerString` that does a strict regex test?

Reply:
Sure, but I think it's Foundry providing that `Number.isNumeric` function. Though I guess it's originally from jQuery.

Alternatively, there is something like `Number.isInteger(Number("10"))`.